containerd-shim-spin-config.md

```diff
@@ -0,0 +1,372 @@
+allowed_outbound_hosts = []
+[component.example.variables]
+kubernetes_service_address = "{{ kubernetes_service_address }}"
+```
+
+Then, it can be accessed in the Spin app using the variables SDK:
+
+```rust
+let addr = variables::get("kubernetes_service_address")?;
+```
+
+In documentation, we should clarify that Spin Wasm components can only be
+configured with Spin application variables. The variable names must be known and
+set in the Spin manifest, as in the above example.
+
+#### Spin Shim Executor Configuration
+
+Users may wish to configure settings for the executors at large, beyond the
+scope of a single application's execution. Containerd supports [configuring
+options for
+runtimes](https://github.com/containerd/containerd/blob/main/docs/man/containerd-config.toml.5.md#multiple-runtimes),
+such as the Spin shim. These options are set in the containerd configuration
+file (`config.toml`), which is where the runtime class manager notifies
+containerd of the existence of the shim. The Runtime Class Manager could also
+add shim options when setting up the shim. For example, it may make the
+following additions to a containerd configuration file:
+
+```toml
+  # Tell containerd where to find the shim (it parses this as /containerd/path/containerd-shim-spin-v2)
+  [plugins."io.containerd.grpc.v1.cri".containerd.runtimes."spin"]
+  runtime_type = "io.containerd.spin.v2"
+  # Add options to each execution of the shim
+  [plugins."io.containerd.grpc.v1.cri".containerd.runtimes."spin".options]
+  Foo = "bar"
+  DISABLE_PRECOMPILATION = "true"
+```
+
+> For an example implementation, see [gVisor's containerd runtime options
+> configuration
+> documentation](https://gvisor.dev/docs/user_guide/containerd/configuration/).
+
+When a container is executed with CRI, these options are sent to Runwasi in a
+[`CreateTaskRequest`](https://github.com/containerd/containerd/blob/main/core/runtime/v2/README.md#container-level-shim-configuration).
+However, right now Runwasi does not parse them. Runwasi should parse the
+options,and pass them to the shim `Engine` through a new `configure_options`
+method that should run prior to any other methods. Runwasi may also extract it's
+own configuration from these containerd runtime options.
+
+The following are Spin shim and Runwasi options that should be configurable:
+
+| Key                             | Example Value | Scope                                                                                                                                                           |
+|---------------------------------|---------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| DISABLE_PRECOMPILATION          | "true"        | shim                                                                                                                                                            |
```

Comment:
what does shim mean here?  `Runwasi` could also read this and disable it at https://github.com/containerd/runwasi/blob/ac4333ed53f246a0ec963c5ecfdef32a73d73a08/crates/containerd-shim-wasm/src/sys/unix/container/instance.rs#L48-L53, this way each shim doesn't need to handle this logic

Reply:
To me, it makes sense that this is piped such that the shim engine returns false to the `can_precompile` call. I also see how runwasi could make the determination as well.